src/tallies/tally_filter.cpp

```diff
@@ -0,0 +1,561 @@
+  {match->bins_.clear();}
+
+  void
+  filter_match_weights_clear(TallyFilterMatch* match)
+  {match->weights_.clear();}
+
+  int
+  filter_match_bins_size(TallyFilterMatch* match)
+  {return match->bins_.size();}
+
+  int
+  filter_match_bins_data(TallyFilterMatch* match, int indx)
+  {return match->bins_[indx-1];}
+
+  double
+  filter_match_weights_data(TallyFilterMatch* match, int indx)
+  {return match->weights_[indx-1];}
+
+  void
+  filter_match_bins_set_data(TallyFilterMatch* match, int indx, int val)
+  {match->bins_[indx-1] = val;}
+
+  TallyFilter*
+  allocate_filter(const char* type)
+  {
+    std::string type_ {type};
+    if (type_ == "azimuthal") {
+      tally_filters.push_back(new AzimuthalFilter());
+    } else if (type_ == "cell") {
+      tally_filters.push_back(new CellFilter());
+    } else if (type_ == "cellborn") {
+      tally_filters.push_back(new CellbornFilter());
+    } else if (type_ == "cellfrom") {
+      tally_filters.push_back(new CellFromFilter());
+    } else if (type_ == "distribcell") {
+      tally_filters.push_back(new DistribcellFilter());
+    } else if (type_ == "energyfunction") {
+      tally_filters.push_back(new EnergyFunctionFilter());
+    } else if (type_ == "legendre") {
+      tally_filters.push_back(new LegendreFilter());
+    } else if (type_ == "material") {
+      tally_filters.push_back(new MaterialFilter());
+    } else if (type_ == "mesh") {
+      tally_filters.push_back(new MeshFilter());
+    } else if (type_ == "meshsurface") {
+      tally_filters.push_back(new MeshSurfaceFilter());
+    } else if (type_ == "mu") {
+      tally_filters.push_back(new MuFilter());
+    } else if (type_ == "polar") {
+      tally_filters.push_back(new PolarFilter());
+    } else if (type_ == "surface") {
+      tally_filters.push_back(new SurfaceFilter());
+    } else if (type_ == "spatiallegendre") {
+      tally_filters.push_back(new SpatialLegendreFilter());
+    } else if (type_ == "sphericalharmonics") {
+      tally_filters.push_back(new SphericalHarmonicsFilter());
+    } else if (type_ == "universe") {
+      tally_filters.push_back(new UniverseFilter());
+    } else if (type_ == "zernike") {
+      tally_filters.push_back(new ZernikeFilter());
+    } else if (type_ == "zernikeradial") {
+      tally_filters.push_back(new ZernikeRadialFilter());
+    } else {
+      return nullptr;
+    }
+    return tally_filters.back();
+  }
+
+  void filter_from_xml(TallyFilter* filt, pugi::xml_node* node)
+  {filt->from_xml(*node);}
+
+  void
+  filter_get_all_bins(TallyFilter* filt, Particle* p, int estimator,
+                      TallyFilterMatch* match)
+  {
+    filt->get_all_bins(p, estimator, *match);
+  }
+
+  void filter_to_statepoint(TallyFilter* filt, hid_t group)
+  {filt->to_statepoint(group);}
+
+  void filter_text_label(TallyFilter* filt, int bin, char* label)
+  {
+    std::string label_str = filt->text_label(bin);
+    int i = 0;
+    for (; i < label_str.size() && i < MAX_LINE_LEN; i++)
```

Comment:
Why do we need this MAX_LINE_LEN check?

Reply:
This function is always called by Fortran with a fixed length character array of `MAX_LINE_LEN+1`